core/src/main/java/com/nuodb/migrator/globalStore/GlobalStore.java

```diff
@@ -0,0 +1,97 @@
+package com.nuodb.migrator.globalStore;
+
+import java.util.*;
+
+import com.google.common.collect.Lists;
+import com.nuodb.migrator.backup.loader.LoadTable;
+import com.nuodb.migrator.jdbc.metadata.Column;
+import com.nuodb.migrator.jdbc.metadata.Schema;
+import com.nuodb.migrator.jdbc.metadata.Table;
+
+import java.util.HashMap;
+
+public class GlobalStore {
+
+    private static GlobalStore instance;
+    private String state;
+
+    private Table table;
+
+    HashMap<String, Boolean> generatedValues = new HashMap<>();
+
+    public void put(String key, boolean value) {
+        generatedValues.put(key, value);
+    }
+
+    public boolean get(String key) {
+        return generatedValues.get(key);
+    }
+
+    private Collection<String> columns = Lists.newArrayList();
+
+    private GlobalStore() {
+        // Private constructor to enforce singleton pattern
+    }
+
+    public static synchronized GlobalStore getInstance() {
+        if (instance == null) {
+            instance = new GlobalStore();
+        }
+        return instance;
+    }
+
+    public synchronized String getState() {
+        return state;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+
+    }
+
+    // helper function to create an alter statement to execute after loading the
+    // data, takes the value from hashmap and compares it.
+    public String alterScript(LoadTable table) {
```

Comment:
Should this method be `static`? The `table` argument hides the `table` instance variable, and it doesn't look like we access any other state in `this` object.